domain-models-runtime/src/main/java/org/folio/rest/impl/JsonSchemasAPI.java

```diff
@@ -0,0 +1,137 @@
+
+public class JsonSchemasAPI implements JsonSchemas {
+
+  private static final Logger log = LoggerFactory.getLogger(JsonSchemasAPI.class);
+
+  private static final Pattern REF_MATCH_PATTERN = Pattern.compile("\\\"\\$ref\\\"\\s*:\\s*\\\"(.*?)\\\"");
+
+  private static final String OKAPI_URL_HEADER = "x-okapi-url";
+  private static final String RAMLS_PATH = System.getProperty("raml_files", GenerateRunner.SOURCES_DEFAULT) + File.separator;
+  private static final String HASH_TAG = "#";
+  private static final String JAR = "jar";
+
+  private static final List<String> JSON_SCHEMAS = getJsonSchemasList();
+
+  @Validate
+  @Override
+  public void getJsonSchemas(
+    String path,
+    Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext
+  ) {
+    vertxContext.runOnContext(v -> {
+      try {
+        if (path == null) {
+          List<String> schemas = getSchemas();
+          asyncResultHandler.handle(
+            Future.succeededFuture(
+              GetJsonSchemasResponse.respond200WithApplicationJson(schemas)
+            )
+          );
+        } else {
+          String okapiUrl = okapiHeaders.get(OKAPI_URL_HEADER);
+          String schema = getJsonSchemaByPath(path, okapiUrl);
+          if (schema != null) {
+            asyncResultHandler.handle(
+              Future.succeededFuture(
+                GetJsonSchemasResponse.respond200WithApplicationSchemaJson(schema)
+              )
+            );
+          } else {
+            String notFoundMessage = "Schema " + path + " not found";
+            asyncResultHandler.handle(
+              Future.succeededFuture(
+                GetJsonSchemasResponse.respond404WithTextPlain(notFoundMessage)
+              )
+            );
+          }
+        }
+      } catch (Exception e) {
+        log.error(e.getMessage(), e);
+        asyncResultHandler.handle(
+          Future.succeededFuture(
+            GetJsonSchemasResponse.respond500WithTextPlain(e.getMessage())
+          )
+        );
+      }
+    });
+  }
+
+  private static List<String> getJsonSchemasList() {
+    URL jsonSchemaListUrl = JsonSchemasAPI.class.getClassLoader().getResource(RAMLS_PATH + GenerateRunner.JSON_SCHEMA_LIST);
+    try {
+      if (jsonSchemaListUrl.toURI().getScheme().equals(JAR)) {
+        return IOUtils.readLines(JsonSchemasAPI.class.getClassLoader().getResourceAsStream(RAMLS_PATH + GenerateRunner.JSON_SCHEMA_LIST), StandardCharsets.UTF_8);
+      } else {
+        return Files.readAllLines(Paths.get(jsonSchemaListUrl.toURI()), StandardCharsets.UTF_8);
+      }
+    } catch (URISyntaxException | IOException e) {
+      return new ArrayList<>();
+    }
+  }
+
+  private List<String> getSchemas() {
+    return JSON_SCHEMAS;
+  }
+
+  private String getJsonSchemaByPath(String path, String okapiUrl) throws URISyntaxException, IOException {
+    URL jsonSchemaUrl = getClass().getClassLoader().getResource(RAMLS_PATH + path);
```

Comment:
Under Windows the values are `RAMLS_PATH + path = ramls\test.schema` and `jsonSchemaUrl = file:/C:/Users/me/raml-module-builder/domain-models-runtime/target/test-classes/ramls%5ctest.schema` and this results in a FileNotFoundException.

Reply:
RMB has ResourceUtil::asString that can handle reading from both jar and classes directory. The latest version from the pull request should work: https://github.com/folio-org/raml-module-builder/pull/325

Reply:
Thanks. Will give that a try.